lib/index.ts

```diff
@@ -0,0 +1,572 @@
+/*
+Copyright 2017 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as request from 'request-promise';
+
+const FrontURL = 'https://api2.frontapp.com';
+
+interface FrontRequest {
+	reqType: string;
+	path: string;
+}
+
+type InternalCallback = (err: Error | null, response: any | null) => void;
+
+export class Front {
+	public comment: {
+		create: (params: FrontCommentRequest.Create, callback?: FrontCallback<FrontComment>) => Promise<FrontComment>;
```

Comment:
I'm not passionate about it, but on balance I'm with @dfunckt, a simple `CommentRequest` is nicer.

Since it's not global you can explicitly namespace it (`import * as Front from 'front'; var x: Front.CommentRequest`) or rename it any time it's ambiguous (`import { CommentRequest as FrontCommentRequest } from 'front'`) we're really just talking about what the _default_ and guessable import name should be, and I think simple works best there, as long as it's in a reasonable context when it's exported.

I'm not sure what you mean about "don't use namespaces" in TS, but you might be thinking of literal namespace declarations, like:

front.ts
```javascript
namespace Front {
  ...
}
```

That's the kind of thing you would do in C#, which is why people warn about it, but if you do that in TS with the standard module per file setup then you get code like `import Front = require('front'); Front.Front.CommentRequest`, and that clearly is a bad idea. That's not what we're doing here though. Is that what you mean? I'd be interested to see arguments against this if you've got any to hand, I can't easily find any myself.